number/number.go

```diff
@@ -0,0 +1,156 @@
+	Number        string
+	Country       string
+	Region        string
+	Locality      string
+	Features      []string
+	Type          []string
+	Status        string
+	SearchPattern string
+}
+
+// request does the exact same thing as Client.Request. It does, however,
+// prefix the path with the Numbers API's root. This ensures the client
+// doesn't "handle" this for us: by default, it uses the REST API.
+func request(c *messagebird.Client, v interface{}, method, path string, data interface{}) error {
+	fmt.Println(path)
+	return c.Request(v, method, fmt.Sprintf("%s/%s", apiRoot, path), data)
+}
+
+// List get all purchased phone numbers
+func List(c *messagebird.Client, listParams *NumberListParams) (*NumberList, error) {
+	uri := GetPath(listParams, pathNumbers)
+
+	numberList := &NumberList{}
+	if err := request(c, numberList, http.MethodGet, uri, nil); err != nil {
+		return nil, err
+	}
+	return numberList, nil
+}
+
+// Search for phone numbers available for purchase
+func Search(c *messagebird.Client, cc string, listParams *NumberListParams) (*NumberSearchingList, error) {
+	uri := GetPath(listParams, pathNumbersAvailable+"/"+cc)
+
+	numberList := &NumberSearchingList{}
+	if err := request(c, numberList, http.MethodGet, uri, nil); err != nil {
+		return nil, err
+	}
+
+	return numberList, nil
+}
+
+// Read get a purchased phone number
+func Read(c *messagebird.Client, phoneNumber string) (*Number, error) {
+	if len(phoneNumber) < 5 {
+		return nil, fmt.Errorf("a phoneNumber is too short")
+	}
+
+	uri := fmt.Sprintf("%s/%s", pathNumbers, phoneNumber)
+
+	number := &Number{}
+	if err := request(c, number, http.MethodGet, uri, nil); err != nil {
+		return nil, err
+	}
+
+	return number, nil
+}
+
+// GetPath get the full path for the request
+func GetPath(listParams *NumberListParams, path string) string {
+	params := paramsForMessageList(listParams)
+	return fmt.Sprintf("%s?%s", path, params.Encode())
+}
+
+// paramsForMessageList build query params
+func paramsForMessageList(params *NumberListParams) *url.Values {
+	urlParams := &url.Values{}
+
+	if params == nil {
+		return urlParams
+	}
+
+	if len(params.Features) > 0 {
+		paramsForArrays("features", "^(sms|voice|mms)$", params.Features, urlParams)
+	}
+
+	if len(params.Type) > 0 {
+		paramsForArrays("type", "^(mobile|mobile|premium_rate)$", params.Type, urlParams)
+	}
+
+	if params.Number != "" {
+		urlParams.Set("number", params.Number)
+	}
+	if params.Country != "" {
+		urlParams.Set("country", params.Country)
+	}
+	if params.Limit != 0 {
+		urlParams.Set("limit", strconv.Itoa(params.Limit))
+	}
+
+	if params.SearchPattern != "" {
+		urlParams.Set("search_pattern", params.SearchPattern)
+	}
+
+	if params.Offset != 0 {
+		urlParams.Set("offset", strconv.Itoa(params.Offset))
+	}
+
+	return urlParams
+}
+
+// paramsForArrays build query for array params
+func paramsForArrays(field string, pattern string, array []string, urlParams *url.Values) {
+	r, _ := regexp.Compile(pattern)
+
```

Comment:
suggestion:
   1.`values`  instead of array would be better name for the 3rd argument of the function 
   2. would prefer range here `for _,value := range values `